dnSpy/dnSpy/Documents/TreeView/DocumentTreeView.cs

```diff
@@ -844,6 +845,38 @@ sealed class AddDocumentInfo {
 			}
 		}
 
+		static void ResolveInternetShortcutFiles(string[] filenames) {
+			for (var i = 0; i < filenames.Length; i++) {
+				var filename = filenames[i];
+				if (!filename.EndsWith(".url", StringComparison.OrdinalIgnoreCase))
+					continue;
+				if (!File.Exists(filename))
+					continue;
+				try {
+					var urlString = GetUrlFromInternetShortcutFile(filename);
+					if (urlString == null)
+						continue;
+					var urlUri = new Uri(urlString);
+					if (urlUri is { IsAbsoluteUri: true, IsFile: true }) {
+						filenames[i] = urlUri.AbsolutePath;
+					}
+				}
+				catch {
+					// ignore any unexpected parsing errors
+				}
+			}
+		}
+
+		// get the URL from a Microsoft [InternetShortcut] .url file
+		static string? GetUrlFromInternetShortcutFile(string filename) {
+			foreach (var line in File.ReadAllLines(filename)) {
+				if (line.StartsWith("URL=")) {
+					return line.Substring(4);
+				}
+			}
+			return null;
+		}
```

Comment:
It seems to me that the `.url` files follow the `.ini` file format. Perhaps instead of iterating over every line in the file to find the key, we could instead use the official way of reason keys from a `.ini` file, the `GetPrivateProfileString` API from `kernel32`.

```c#
[DllImport("kernel32", CharSet = CharSet.Unicode)]
static extern int GetPrivateProfileString(string lpAppName, string lpKeyName, string lpDefault, StringBuilder lpReturnedString, int nSize, string lpFileName);
```

```c#
var urlValue = new StringBuilder(255);
int charsCopied = GetPrivateProfileString("InternetShortcut", "URL", "", urlValue, 255, filename);
urlValue.ToString() // the value of the URL key
```

If the `charsCopied` variable is not positive, we can return null as it means that the `GetPrivateProfileString` function has failed.

The comment on line 870 is redundant in my opinion.

Reply:
I don't like relying on Windows specific APIs, but I think that ship has sailed in dnSpy.
Anyway, created a dedicated static class to hold the `GetUrlFromInternetShortcutFile` and `GetPrivateProfileString` methods.